### PR TITLE
Merge Launcher into Main app

### PR DESCRIPTION
**Pull request by agarg8487**
|PR checklist|Status|
|-----|:------:|
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|Yes|
|Is code analysis task executed and issues resolved in the CPP/native code?|NA|
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|No|
|Is code commenting done properly for the changes made?|Yes|
|Is correct variable naming convention followed?|Yes|
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|Yes|
|Are the message strings used in the application taken from strings.xml file?|No|
|Are the permissions which are not required removed from AndroidManifest.java file?|NA|
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA|
|Are the generated files correctly configured in .gitignore?|Yes|
|Are the release signing configurations removed from the gradle files?|NA|
|Is commented code removed from all files?|Yes|
|Are updated images compressed (< 100 KB)?|NA|
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|Yes|

### CODE REVIEW

File: kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitlogger/KitKitLogger.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ public KitKitLogger(String appName, Context ctnx) {
 
 
 //        try {
-//            Context launcherContext = ctnx.createPackageContext("com.maq.xprize.kitkitlauncher.hindi", Context.MODE_PRIVATE);
+//            Context launcherContext = ctnx.createPackageContext("org.cocos2dx.cpp.kitkitlauncher.hindi", Context.MODE_PRIVATE);
 //            SharedPreferences prefs = launcherContext.getSharedPreferences("sharedpreference",Context.MODE_PRIVATE);
 //            installId = prefs.getString("installId","");
 //        } catch (PackageManager.NameNotFoundException e) {
```
Comment (ankitk20) [inline kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitlogger/KitKitLogger.java]: Remove commented code

**Review by harry6396 — DISMISSED** [review]

File: kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitProvider/User.java [hunk 2]
```diff
@@ -78,97 +78,136 @@ public User(String productname, int stars) {
...
     }
 
     public void setNumStars(int stars) {
         this._numStars = stars;
     }
 
-    public int getNumStars() {
-        return this._numStars;
+    public boolean isFinishTutorial() {
+        return true; /*this._finishTutorial;*/
```
Comment (ankitk20) [inline kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitProvider/User.java]: Remove this

File: kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitlogger/KitKitLoggerActivity.java [hunk 3]
```diff
@@ -62,30 +63,29 @@ protected void onCreate(Bundle savedInstanceState) {
...
-            Log.v(TAG,"Permission: "+permissions[0]+ "was "+grantResults[0]);
+        if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            Log.v(TAG, "Permission: " + permissions[0] + "was " + grantResults[0]);
         }
     }
 
     @Override
     protected void attachBaseContext(Context newBase) {
         try {
             //Context launcherContext = newBase.createPackageContext("todoschoollauncher.enuma.com.todoschoollauncher",0);
```
Comment (ankitk20) [inline kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitlogger/KitKitLoggerActivity.java]: Remove this line

File: mainapp/proj.android-studio/app/build.gradle [hunk 4]
```diff
@@ -141,4 +141,18 @@ dependencies {
 //    englishCompile project(path: ':kitkitlogger', configuration: 'englishRelease')
 //    swahiliCompile project(path: ':kitkitlogger', configuration: 'swahiliRelease')
     compile 'com.android.support.constraint:constraint-layout:1.1.3'
+//    Used by Launcher
+    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
+        exclude group: 'com.android.support', module: 'support-annotations'
+    })
```
Comment (ankitk20) [inline mainapp/proj.android-studio/app/build.gradle]: Remove test libs

File: mainapp/proj.android-studio/app/build.gradle [hunk 5]
```diff
@@ -141,4 +141,18 @@ dependencies {
...
+    compile 'com.android.support:appcompat-v7:25.1.0'
+    compile 'com.android.support:design:25.1.0'
```
Comment (ankitk20) [inline mainapp/proj.android-studio/app/build.gradle]: Upgrade this to 28.0.0

File: mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java [hunk 6]
```diff
@@ -102,7 +102,10 @@ public void run() {
                     @Override
                     public void run() {
                         try {
+//                            Calling Kitkit Launcher MainActivity
                             _activity.moveTaskToBack(true);
+                            Intent startLauncher = new Intent(_activity, org.cocos2dx.cpp.kitkitlauncher.hindi.MainActivity.class);
+                            _activity.startActivity(startLauncher);
                         } catch (Exception e) {
                             Log.e(TAG, "", e);
```
Comment (ankitk20) [inline mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java]: Remove log

File: mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java [hunk 7]
```diff
@@ -423,7 +424,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
         // init sign-language value
         try {
-            Context launcherContext = createPackageContext("com.maq.xprize.kitkitlauncher.hindi", 0);
+            Context launcherContext = createPackageContext("com.maq.xprize.kitkitschool.hindi", 0);
```
Comment (ankitk20) [inline mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java]: Use method to get package name.

File: mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/ReadingBird/Recognizer.java [hunk 8]
```diff
@@ -412,16 +446,16 @@ public void run() {
                             }
 
                             if (mbCheckVolume == true) {
```
Comment (ankitk20) [inline mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/ReadingBird/Recognizer.java]: Do not use == true

File: mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/ReadingBird/Recognizer.java [hunk 9]
```diff
@@ -412,16 +446,16 @@ public void run() {
...
-                                Log.i("-> Recognizer record " + 0 +  ", value : " + Math.abs(toTransform[0]) + ", " + ((SAMPLE_RATE / 2.0f / Recognizer.this.bufferSize) * 0) + "Hz");
+                                Log.i("Recognizer record " + maxIndex + ", value : " + maxValue + ", " + ((SAMPLE_RATE / 2.0f / Recognizer.this.bufferSize) * maxIndex) + "Hz");
+                                Log.i("-> Recognizer record " + 0 + ", value : " + Math.abs(toTransform[0]) + ", " + ((SAMPLE_RATE / 2.0f / Recognizer.this.bufferSize) * 0) + "Hz");
                                 if (mTriggerVolume <= volume && Math.abs(toTransform[0]) < 10) {
                                     mbCheckVolume = false;
                                 }
 
                             }
 
                             if (mbCheckVolume == false) {
```
Comment (ankitk20) [inline mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/ReadingBird/Recognizer.java]: Do not use == false

**Review by ankitk20 — CHANGES_REQUESTED** [review]

**Review by harry6396 — APPROVED** [review]

**Review by ankitk20 — APPROVED** [review]

**Review by mathewnik90 — APPROVED** [review]

**Pull request merged**